pkg/controllers/dnspolicy/dnspolicy_dnsrecords.go

```diff
@@ -0,0 +1,118 @@
+package dnspolicy
+
+import (
+	"context"
+	"fmt"
+
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	crlog "sigs.k8s.io/controller-runtime/pkg/log"
+	gatewayv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
+
+	"github.com/kuadrant/kuadrant-operator/pkg/reconcilers"
+
+	"github.com/Kuadrant/multicluster-gateway-controller/pkg/apis/v1alpha1"
+	"github.com/Kuadrant/multicluster-gateway-controller/pkg/traffic"
+)
+
+func (r *DNSPolicyReconciler) reconcileDNSRecords(ctx context.Context, dnsPolicy *v1alpha1.DNSPolicy, gwDiffObj *reconcilers.GatewayDiff) error {
+	log := crlog.FromContext(ctx)
+
+	for _, gw := range gwDiffObj.GatewaysWithInvalidPolicyRef {
+		log.V(1).Info("reconcileDNSRecords: gateway with invalid policy ref", "key", gw.Key())
+		//ToDo Since gateways own DNSRecords I don't think there is anything to do here?
```

Comment:
should we remove the policy ref or is that responsibility somewhere else?

Reply:
I assume if there is no policy then a new default is created?

Reply:
> should we remove the policy ref or is that responsibility somewhere else?

Policy refs are removed elsewhere at the end of the policy reconcile. This one is focused on dnsrecord resources that are created. If we wanted records to be removed when a policy no longer applies, we would do it here.

> I assume if there is no policy then a new default is created?

The dnspolicy reconciler will not create policies. We currently have the gateway controller creating a default one, but if it is removed it would need to be manually created again re. https://github.com/Kuadrant/multicluster-gateway-controller/pull/237#discussion_r1224293911